core-services/src/main/java/org/zalando/nakadi/service/publishing/EventOwnerExtractorFactory.java

```diff
@@ -24,37 +24,78 @@ public EventOwnerExtractorFactory(final FeatureToggleService featureToggleServic
         this.featureToggleService = featureToggleService;
     }
 
-    public Function<JSONObject, EventOwnerHeader> createExtractor(final EventType eventType) {
+    public EventOwnerExtractor createExtractor(final EventType eventType) {
         final EventOwnerSelector selector = eventType.getEventOwnerSelector();
         if (null == selector || !(featureToggleService.isFeatureEnabled(Feature.EVENT_OWNER_SELECTOR_AUTHZ))) {
             return null;
         }
         switch (selector.getType()) {
             case PATH:
-                return createPathExtractor(selector);
+                return createPathExtractor(selector.getName(), selector.getValue());
             case STATIC:
-                return createStaticExtractor(selector);
+                return createStaticExtractor(selector.getName(), selector.getValue());
+            case METADATA:
+                if (!"event_owner".equals(selector.getValue())) {
```

Comment:
I think that check should be earlier, when the event type is created. Otherwise it creates the situation when the user create wrong event owner selector for the event type and can not change it. I suggest to disallow providing value for the metadata selector type.

Reply:
See: https://github.com/zalando/nakadi/pull/1409/commits/2236d1a99c4ebf8fc33723fb24d9283f9e355616